### PR TITLE
Manifest: new credits.note property, intended to be localized

### LINKED ISSUE #2172: Localize addon credits
**Is your feature request related to a problem? Please describe.**

Some addon credits should be localized, for example *Boomer001 (Userstyle)* in `forum-quote-code-beautifier` and *Maximouse (addon, "Experimental" Dark)* in `editor-dark-mode`.

**Describe the solution you'd like**

I think the best solution is to add another property to credits, separate from `name` (maybe `contribution`), that would be localizable. For example:
```json
{
  "name": "Maximouse",
  "contribution": "addon, \"Experimental\" Dark",
  "link": "..."
}
```

Alternatively, the `name` property could be localized.

### PR DESCRIPTION
**Pull request by WorldLanguages**
Resolves #2172

Also, moves the Silk credits from forum-toolbar from the description to the addon credits.

### CODE REVIEW

**Comment by mxmou** [comment]
Related: #2172

**Comment by WorldLanguages** [comment]
@mxmou I think it would be easy to do that, so let's do it now! I'll convert this PR to a fix for that.

**Comment by WorldLanguages** [comment]
Hmm, credits would need to have an ID if they have a note.

**Comment by RedGuy12** [comment]
(Note to self to add credit tooltips to scradd when this is done)

**Comment by WorldLanguages** [comment]
TODOs:

- [ ] Manifest schema
- [ ] Update localization script

**Comment by RedGuy12** [comment]
@WorldLanguages update fuse settings so this can be searchable?

**Comment by WorldLanguages** [comment]
@RedGuy12 Doesn't make sense IMO.

**Comment by RedGuy12** [comment]
Well we already let credit.name be searchable and this info was in credit.name before. so removing it could break some old behaviour as well but that's really my main reason tbh

File: addons/forum-quote-code-beautifier/addon.json [hunk 1]
```diff
@@ -3,10 +3,14 @@
...
     {
-      "name": "GrahamSH (Port)"
+      "name": "GrahamSH",
+      "note": "addon",
+      "id": "addon"
     },
     {
-      "name": "Boomer001 (Userstyle)"
+      "name": "Boomer001",
+      "note": "userstyle",
```
Comment (RedGuy12) [inline addons/forum-quote-code-beautifier/addon.json:12]: while we're at it, change this to "original userstyle"?

File: addons/forum-quote-code-beautifier/addon.json [hunk 2]
```diff
@@ -3,10 +3,14 @@
   "description": "Shows nicer code and quote blocks on the forums.",
   "credits": [
     {
-      "name": "GrahamSH (Port)"
+      "name": "GrahamSH",
+      "note": "addon",
```
Comment (RedGuy12) [inline addons/forum-quote-code-beautifier/addon.json:7]: "port" sounds better imo

**Comment by RedGuy12** [comment]
live-featured-project, material-forum-editor-buttons, scratch-notifier still have unlocalized notes

File: addons/forum-quote-code-beautifier/addon.json (hunk 2 above)
Comment (WorldLanguages) [reply addons/forum-quote-code-beautifier/addon.json:7]: port, where ships load? tcp ports?

Comment (RedGuy12) [reply addons/forum-quote-code-beautifier/addon.json:7]: hmm i see your point. maybe `port to addon`?

Comment (mxmou) [reply addons/forum-quote-code-beautifier/addon.json:7]: "Addon version"?

**Comment by apple502j** [comment]
l10n: ScratchAddons/l10n-script#10
Manifest: ScratchAddons/manifest-schema#55

**Review by apple502j — APPROVED** [review]

File: addons/forum-quote-code-beautifier/addon.json (hunk 2 above)
Comment (WorldLanguages) [reply addons/forum-quote-code-beautifier/addon.json:7]: I'm going with "addon version"

File: addons/forum-quote-code-beautifier/addon.json (hunk 1 above)
Comment (WorldLanguages) [reply addons/forum-quote-code-beautifier/addon.json:12]: Okay

File: background/load-addon-manifests.js [hunk 3]
```diff
@@ -21,6 +21,11 @@
           info.text = scratchAddons.l10n.get(`${folderName}/@info-${info.id}`, {}, info.text);
         }
       }
+      if (manifest.credits) {
+        for (const credit of manifest.credits || []) {
+          credit.note = scratchAddons.l10n.get(`${folderName}/@credits-${credit.id}`, {}, credit.note);
```
Comment (RedGuy12) [inline background/load-addon-manifests.js:26]: This does not guarantee credit.id is set. If it is not set, credit.note ends up addonName/@credits-undefined.

Comment (apple502j) [reply background/load-addon-manifests.js:26]: @RedGuy12 Such errors should be caught by the schema validator.

File: addons/scratch-notifier/addon.json [hunk 4]
```diff
@@ -9,7 +9,9 @@
       "name": "NitroCipher/ZenithRogue"
     },
     {
-      "name": "notificationsounds.com (Scratch Addons ping)"
+      "name": "notificationsounds.com",
+      "note": "Scratch Addons ping",
+      "id": "Scratch Addons ping"
```
Comment (RedGuy12) [inline addons/scratch-notifier/addon.json:14]: does this work with non-alphanumeric characters?

File: background/load-addon-manifests.js (hunk 3 above)
Comment (RedGuy12) [reply background/load-addon-manifests.js:26]: either both `id` and `note` is set, or neither is set. if both are set, credit.note gets reassigned to the message `addonName/@credits-creditId`. If neither are set, **this line still runs**. We forgot to add any conditional here. it tries to fetch the string with id `addonName/@credits-undefined`.  No string is found so it falls back to the key, `"addonName/@credits-undefined"`.

File: addons/scratch-notifier/addon.json (hunk 4 above)
Comment (WorldLanguages) [reply addons/scratch-notifier/addon.json:14]: Yeah this was unintentional

**Pull request merged**